crates/holochain_cascade/src/lib.rs

```diff
@@ -623,27 +624,25 @@ where
         options: GetOptions,
     ) -> CascadeResult<Option<EntryDetails>> {
         let authoring = self.am_i_authoring(&entry_hash.clone().into())?;
-        let authority = self.am_i_an_authority(entry_hash.clone().into()).await?;
+        let authority = authoring || self.am_i_an_authority(entry_hash.clone().into()).await?;
         let query: GetEntryDetailsQuery = self.construct_query_with_data_access(entry_hash.clone());
 
-        // We don't need metadata and only need the content
-        // so if we have it locally then we can avoid the network.
-        if let GetStrategy::Content = options.strategy {
-            let results = self.cascading(query.clone()).await?;
-            // We got a result so can short circuit.
-            if results.is_some() {
-                return Ok(results);
-            // We didn't get a result so if we are either authoring
-            // or the authority there's nothing left to do.
-            } else if authoring || authority {
-                return Ok(None);
+        match options.strategy {
+            GetStrategy::Content => {
+                // We don't need metadata and only need the content
+                // so if we have it locally then we can avoid the network.
+                let results = self.cascading(query.clone()).await?;
+                if results.is_some() || authority {
+                    return Ok(results);
+                } else {
+                    self.fetch_record(entry_hash.into(), options.into()).await?;
+                }
+            }
+            GetStrategy::Latest => {
+                if !authoring {
```

Comment:
How can someone request an entry that I'm authoring? Has that not _not_ been published yet?

Reply:
If I'm `authoring`, that means this cascade call is happening *during* the authoring process

Reply:
How can someone know the hash of an entry and request it while the entry isn't published yet?